File: Vision/Barcode Reader/Java/src/main/java/frc/robot/subsystems/VisionSubsystem.java
package frc.robot.subsystems;

import edu.wpi.first.networktables.NetworkTable;
import edu.wpi.first.networktables.NetworkTableEntry;
import edu.wpi.first.networktables.NetworkTableInstance;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
import edu.wpi.first.wpilibj2.command.SubsystemBase;

public class VisionSubsystem extends SubsystemBase
{
    private NetworkTableInstance inst = NetworkTableInstance.getDefault();
    private NetworkTable table = inst.getTable("Vision");
    private NetworkTableEntry data;

    public VisionSubsystem()
    {
        SmartDashboard.putBoolean("Get New Barcode", false);
    }

    public void readBarcode()
    {
        table.getEntry("readBarcode").setBoolean(true);
    }

    public void printBarcode()
    {
        data = table.getEntry("barcodeData");
        SmartDashboard.putString("Barcode Data", data.getString("Nothing was read"));
    }

    @Override
    public void periodic()
    {
        printBarcode();
    }
}

File: Vision/Barcode Reader/C++/src/main/cpp/subsystems/VisionSubsystem.cpp
#include "subsystems/VisionSubsystem.h"

VisionSubsystem::VisionSubsystem()
{
    auto inst = nt::NetworkTableInstance::GetDefault();
    auto table = inst.GetTable("Vision");
    data = table->GetEntry("barcodeData");
    newBarcode = table->GetEntry("readBarcode");
    frc::SmartDashboard::PutBoolean("Get New Barcode", false);
}

void VisionSubsystem::ReadBarcode()
{
    newBarcode.SetBoolean(true);
}

void VisionSubsystem::PrintBarcode()
{
    frc::SmartDashboard::PutString("Barcode Data", data.GetString("Nothing was read"));
}

void VisionSubsystem::Periodic()
{
    PrintBarcode();
}